File: ReadWrite/ReadWrite.cpp

#include <fstream>
#include <iostream>
#include <string>

using namespace std;

int main()
{
    fstream file("numbers.txt", ios::in);
    string line;
    int sum = 0;


    if (!file.is_open()) {
        cerr << "err: cannot open file " << endl;
        return 1;
    }


    while (getline(file, line)) {
        cout << line << endl;
        sum += stoi(line);
    }

    file.close();
    
    cout << "sum is: " << sum << endl;
}

File: python/read_write.py
path = "../data/numbers.txt"

numbers = []

with open(path, "r") as file: 
    for line in file:
        numbers.append(int(line))

print(numbers)

out_path = "../data/numbers_decreased.txt"

with open(out_path, "w") as file:
    for number in numbers:
        line = str(number - 100) + "\n"
        file.write(line)


